### ISSUE TITLE
Energy Meter data gets stuck

### ISSUE THREAD

**Issue by SWarnberg** [ricott/sma.modbus#23]
Hi

I've connected an Energy Meter 2.0 to the app, and initially the data comes in fine. However, after some minutes the data gets stuck and the same values is shown forever in the Homey app. See picture below.

![image](https://user-images.githubusercontent.com/26246346/229295506-1ca5c989-6c98-48d4-8134-f7937885eef8.png)

After a restart of the app, it takes 7 minutes before it gets stuck again:

![image](https://user-images.githubusercontent.com/26246346/229295732-aeecdfea-5913-4773-9fbc-2dbe696f175f.png)

Any idea of the cause and solution?
I've sent a diagnostics report as well from the app. The code to it is 8aa229c7-5dc4-44be-bf24-b8d7f7c063dc.

Staffan

**Comment by SWarnberg** [comment]
Hi @ricott 

Some more details. I ran the code and find that the messages stops coming in on the client. This happens every time after 4m20s. There is no 'error' or 'close' message coming in either.

```
function initListenersAndConnect(self) {

    self.client = dgram.createSocket({ type: 'udp4' }); //, reuseAddr: true });

    self.client.on("message", function (message, rinfo) {
```

**Comment by SWarnberg** [comment]
I added a PR #24, which is a workaround that restarts the client when messages are lost. If there's no proper network solution, then this could work. Tested in my environment for 40hrs w/o any issues.

**Comment by Mark323513** [comment]
I have the same issue with mz Homey 2023,

The diagnostic report has no 09307C63

**Comment by SWarnberg** [comment]
@Mark323513 You can clone my PR #24 and install it on your Homey. I've run it for almost two months now without any issues.
This repo seems to be idle right now.

**Comment by ricott** [comment]
Thinking out loud. Since the EM2 device is consuming UDP multicast datagrams simply by opening a port locally on the Homey - it doesn't connect to anything remote. Could these problems be based on your Homey disconnecting from your wifi from time to time and this is what causes the problem?

**Comment by SWarnberg** [comment]
I wouldn't think so, since this happens exactly after 4+ minutes, every time, while I'm still connected to Homey in debug mode (or with app)..

**Comment by ricott** [comment]
Historically, over the years I have developed the app, the only problems related to EM2 have been network related. Since the logic is so simple, no stateful connectivity to any external device, just open local port and accept what is pushed there.

There are 1000+ installs of the app, so I still think what you are experiencing is some local network wifi/UDP/multicast issue - otherwise a lot of users would have complained. Anyhow, I have merged your pull request.
